From f169108bcee5d71677730d8dc351309d31226be8 Mon Sep 17 00:00:00 2001
From: Lyon Wang <lyon.wang@freescale.com>
Date: Wed, 4 Nov 2015 18:16:52 +0800
Subject: [PATCH 03/13] unset FLAG_DISCONT when push to adapter

Since DISCONT flag will be handled by priv->discont flag.
There is no need to copy DISCONT flag to adapter.
Otherwise, the DISCONT flag will alwasys been set on each frame,
for the first chain in buffer.

Upstream-Status: Pending
https://bugzilla.gnome.org/show_bug.cgi?id=757575

Signed-off-by: Lyon Wang <lyon.wang@freescale.com>
(cherry picked from commit 0f3f083e32a8919d30c946d82cf700fd8a1c6cac)
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 libs/gst/base/gstbaseparse.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libs/gst/base/gstbaseparse.c b/libs/gst/base/gstbaseparse.c
index 68db103..e9fd4e1 100644
--- a/libs/gst/base/gstbaseparse.c
+++ b/libs/gst/base/gstbaseparse.c
@@ -3253,6 +3253,7 @@ gst_base_parse_chain (GstPad * pad, GstObject * parent, GstBuffer * buffer)
         parse->priv->discont = TRUE;
       }
     }
+    GST_BUFFER_FLAG_UNSET(buffer, GST_BUFFER_FLAG_DISCONT);
     gst_adapter_push (parse->priv->adapter, buffer);
   }
 
-- 
2.49.0

